hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -128,6 +137,33 @@ public HoodieFileGroupReader(HoodieReaderContext<T> readerContext,
     this.readerState.baseFileAvroSchema = avroSchema;
     this.readerState.logRecordAvroSchema = avroSchema;
     this.readerState.mergeProps.putAll(props);
+
+    FileSystem fs = readerContext.getFs(logFilePathList.get().get(0), hadoopConf);
+    HoodieTableMetaClient metaClient = HoodieTableMetaClient
+        .builder().setConf(fs.getConf()).setBasePath(tablePath).build();
+
+    HoodieFileGroupRecordBuffer keyBasedRecordBuffer = new HoodieKeyBasedFileGroupRecordBuffer(
+        readerContext,
+        readerState.logRecordAvroSchema,
+        readerState.baseFileAvroSchema,
+        Option.of(getRelativePartitionPath(
+            new Path(readerState.tablePath), new Path(logFilePathList.get().get(0)).getParent())),
+        Option.of(metaClient.getTableConfig().getPartitionFields()),
+        recordMerger,
+        props,
+        metaClient);
+    HoodieFileGroupRecordBuffer positionBasedRecordBuffer = new HoodiePositionBasedFileGroupRecordBuffer(
+        readerContext,
```

Comment:
Can we move the record buffer instance creation to it's factory class? There is no need to create 2 instantces altogether.

Reply:
Move it out of this this class.